variants/github_actions_ci/workflows/deploy_to_ec2.yml

```diff
@@ -0,0 +1,72 @@
+name: deploy_to_ec2
+on:
+  workflow_call:
+    inputs:
+      environment:
+        type: string
+        required: true
+      assume_role_arn:
+        type: string
+        required: true
+      aws_region:
+        type: string
+        required: true
+      timeout_minutes:
+        type: number
+        required: false
+        default: 15
+      slack_icon:
+        type: string
+        required: false
+        default: https://github.com/ackama.png?size=48
+      slack_channel:
+        type: string
+        required: false
+        default:
+      slack_username:
+        type: string
+        required: false
+        default: "Github Actions CI"
+    secrets:
+      ssh_private_key:
+        required: true
+      slack_webhook:
+        required: false
+jobs:
+  deploy_to_ec2_via_oidc:
+    name: Deploy to AWS EC2 with Capistrano, authenticated by Github OIDC
+    runs-on: ubuntu-latest
+    timeout-minutes: ${{ inputs.timeout_minutes }}
+    permissions:
+      # These permissions are needed to interact with GitHub's OIDC Token
+      # endpoint.
+      id-token: write
+      contents: read
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ruby/setup-ruby@v1
+        with:
+          bundler-cache: true
+      - name: Configure AWS credentials using OpenID Connect to AssumeRole
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ inputs.assume_role_arn }}
+          aws-region: ${{ inputs.aws_region }}
+      - name: Install SSH key for production
+        uses: shimataro/ssh-key-action@v2
+        with:
+          key: ${{ secrets.ssh_private_key }}
+          known_hosts: unnecessary
+      - name: Deploy application
+        run: bundle exec cap ${{ inputs.environment }} deploy
+      - name: Report to Slack
+        if: ${{ secrets.slack_webhook }}
+        uses: rtCamp/action-slack-notify@v2
+        env:
+          SLACK_WEBHOOK: ${{ secrets.slack_webhook }}
+          SLACK_CHANNEL: ${{ inputs.slack_channel }}
+          SLACK_ICON: ${{ inputs.slack_icon }}
+          SLACK_USERNAME: ${{ inputs.slack_username }}
+          SLACK_FOOTER: ""
+          SLACK_COLOR: ${{ job.status }}
+          SLACK_MESSAGE: "${{ format('{0} job result: {1}', github.job, job.status) }}"
```

Comment:
This isn't tested yet. I'll do it for real on a project during the week and update here if required